src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs

```diff
@@ -44,8 +43,12 @@ public class MvcRazorHost : RazorEngineHost, IMvcRazorHost
         /// <param name="root"/>.
         /// </summary>
         /// <param name="root">The path to the application base.</param>
+        // Note: This constructor is used by tooling and is created once for each
+        // Razor page that is loaded. Consequently, each loaded page has its own copy of
+        // the CodeTreeCache, but this ok - having a shared CodeTreeCache per application in tooling
+        // is problematic to manage.
         public MvcRazorHost(string root) :
-            this(new PhysicalFileProvider(root))
+            this(new CodeTreeCache(new PhysicalFileProvider(root)))
```

Comment:
The way this works in tooling is that they create one instance of MvcRazorHost per open Razor file open. Consequently we'll get _ViewStart caching on a per file basis. Changing it to be per project requires more effort on their end to manage what project a file belongs to which seemed problematic.